// fujitsu-xfs4iot-api/dist/xfs4iot-interface/Common/InterfaceNameEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterfaceNameEnum = void 0;
var InterfaceNameEnum;
(function (InterfaceNameEnum) {
    /** Common interface. Every device implements this interface. */
    InterfaceNameEnum["COMMON"] = "common";
    /** CardReader interface. */
    InterfaceNameEnum["CARD_READER"] = "cardReader";
    /** CashAcceptor interface. */
    InterfaceNameEnum["CASH_ACCEPTOR"] = "cashAcceptor";
    /** CashDispenser interface. */
    InterfaceNameEnum["CASH_DISPENSER"] = "cashDispenser";
    /** CashManagement interface. */
    InterfaceNameEnum["CASH_MANAGEMENT"] = "cashManagement";
    /** PinPad interface. */
    InterfaceNameEnum["PIN_PAD"] = "pinPad";
    /** Crypto interface. */
    InterfaceNameEnum["CRYPTO"] = "crypto";
    /** KeyManagement interface. */
    InterfaceNameEnum["KEY_MANAGEMENT"] = "keyManagement";
    /** Keyboard interface. */
    InterfaceNameEnum["KEYBOARD"] = "keyboard";
    /** TextTerminal interface. */
    InterfaceNameEnum["TEXT_TERMINAL"] = "textTerminal";
    /** Printer interface. */
    InterfaceNameEnum["PRINTER"] = "printer";
    /** BarcodeReader interface. */
    InterfaceNameEnum["BARCODE_READER"] = "barcodeReader";
    /** Lights interface. */
    InterfaceNameEnum["LIGHTS"] = "lights";
    /** Auxiliaries interface. */
    InterfaceNameEnum["AUXILIARIES"] = "auxiliaries";
    /** VendorMode interface. */
    InterfaceNameEnum["VENDOR_MODE"] = "vendorMode";
    /** VendorApplication interface. */
    InterfaceNameEnum["VENDOR_APPLICATION"] = "vendorApplication";
    /** Storage interface */
    InterfaceNameEnum["STORAGE"] = "storage";
})(InterfaceNameEnum = exports.InterfaceNameEnum || (exports.InterfaceNameEnum = {}));
