// fujitsu-xfs4iot-api/dist/xfs4iot-interface/CashManagement/unsolicited/CashManagementUnsolicitedItemsPresentedEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemsPresentedEventPayloadPositionEnum = void 0;
var ItemsPresentedEventPayloadPositionEnum;
(function (ItemsPresentedEventPayloadPositionEnum) {
    /** Default input position. */
    ItemsPresentedEventPayloadPositionEnum["IN_DEFAULT"] = "inDefault";
    /** Left input position. */
    ItemsPresentedEventPayloadPositionEnum["IN_LEFT"] = "inLeft";
    /** Right input position. */
    ItemsPresentedEventPayloadPositionEnum["IN_RIGHT"] = "inRight";
    /** Center input position. */
    ItemsPresentedEventPayloadPositionEnum["IN_CENTER"] = "inCenter";
    /** Top input position. */
    ItemsPresentedEventPayloadPositionEnum["IN_TOP"] = "inTop";
    /** Bottom input position. */
    ItemsPresentedEventPayloadPositionEnum["IN_BOTTOM"] = "inBottom";
    /** Front input position. */
    ItemsPresentedEventPayloadPositionEnum["IN_FRONT"] = "inFront";
    /** Rear input position. */
    ItemsPresentedEventPayloadPositionEnum["IN_REAR"] = "inRear";
    /** Default output position. */
    ItemsPresentedEventPayloadPositionEnum["OUT_DEFAULT"] = "outDefault";
    /** Left output position. */
    ItemsPresentedEventPayloadPositionEnum["OUT_LEFT"] = "outLeft";
    /** Right output position. */
    ItemsPresentedEventPayloadPositionEnum["OUT_RIGHT"] = "outRight";
    /** Center output position. */
    ItemsPresentedEventPayloadPositionEnum["OUT_CENTER"] = "outCenter";
    /** Top output position. */
    ItemsPresentedEventPayloadPositionEnum["OUT_TOP"] = "outTop";
    /** Bottom output position. */
    ItemsPresentedEventPayloadPositionEnum["OUT_BOTTOM"] = "outBottom";
    /** Front output position. */
    ItemsPresentedEventPayloadPositionEnum["OUT_FRONT"] = "outFront";
    /** Rear output position. */
    ItemsPresentedEventPayloadPositionEnum["OUT_REAR"] = "outRear";
})(ItemsPresentedEventPayloadPositionEnum = exports.ItemsPresentedEventPayloadPositionEnum || (exports.ItemsPresentedEventPayloadPositionEnum = {}));
