// fujitsu-xfs4iot-api/dist/xfs4iot-interface/CashManagement/unsolicited/CashManagementUnsolicitedItemsTakenEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemsTakenEventPayloadPositionEnum = void 0;
var ItemsTakenEventPayloadPositionEnum;
(function (ItemsTakenEventPayloadPositionEnum) {
    /** Default input position. */
    ItemsTakenEventPayloadPositionEnum["IN_DEFAULT"] = "inDefault";
    /** Left input position. */
    ItemsTakenEventPayloadPositionEnum["IN_LEFT"] = "inLeft";
    /** Right input position. */
    ItemsTakenEventPayloadPositionEnum["IN_RIGHT"] = "inRight";
    /** Center input position. */
    ItemsTakenEventPayloadPositionEnum["IN_CENTER"] = "inCenter";
    /** Top input position. */
    ItemsTakenEventPayloadPositionEnum["IN_TOP"] = "inTop";
    /** Bottom input position. */
    ItemsTakenEventPayloadPositionEnum["IN_BOTTOM"] = "inBottom";
    /** Front input position. */
    ItemsTakenEventPayloadPositionEnum["IN_FRONT"] = "inFront";
    /** Rear input position. */
    ItemsTakenEventPayloadPositionEnum["IN_REAR"] = "inRear";
    /** Default output position. */
    ItemsTakenEventPayloadPositionEnum["OUT_DEFAULT"] = "outDefault";
    /** Left output position. */
    ItemsTakenEventPayloadPositionEnum["OUT_LEFT"] = "outLeft";
    /** Right output position. */
    ItemsTakenEventPayloadPositionEnum["OUT_RIGHT"] = "outRight";
    /** Center output position. */
    ItemsTakenEventPayloadPositionEnum["OUT_CENTER"] = "outCenter";
    /** Top output position. */
    ItemsTakenEventPayloadPositionEnum["OUT_TOP"] = "outTop";
    /** Bottom output position. */
    ItemsTakenEventPayloadPositionEnum["OUT_BOTTOM"] = "outBottom";
    /** Front output position. */
    ItemsTakenEventPayloadPositionEnum["OUT_FRONT"] = "outFront";
    /** Rear output position. */
    ItemsTakenEventPayloadPositionEnum["OUT_REAR"] = "outRear";
})(ItemsTakenEventPayloadPositionEnum = exports.ItemsTakenEventPayloadPositionEnum || (exports.ItemsTakenEventPayloadPositionEnum = {}));
